File: 卡码/代码随想录/1/09-动态规划/C++/dp/03-打家劫舍/LeetCode_0198.cpp
//198. 打家劫舍
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int rob(vector<int>& nums) {
		int dp[nums.size()];
		dp[0]=nums[0];
		if(nums.size()>1) dp[1]=max(nums[0],nums[1]);
		for(int i=2;i<nums.size();i++){
			dp[i]=max(dp[i-2]+nums[i],dp[i-1]);
		}
		return dp[nums.size()-1];
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/09-动态规划/Python/03-打家劫舍/LeetCode_0198.py
#198. 打家劫舍
from typing import *

class Solution:
  def rob(self, nums: List[int]) -> int:
    n:int=len(nums)
    dp:List[int]=[0]*(n)
    dp[0]=nums[0]
    if n>1:
      dp[1]=max(nums[0],nums[1])
    for i in range(2,n):
      dp[i]=max(dp[i-1],dp[i-2]+nums[i])
    return dp[n-1]